### PR TITLE
JIT: extract BBJ_COND to BBJ_ALWAYS profile repair as utility

### PR DESCRIPTION
**Pull request by AndyAyersMS**
and call it from a few places.

### CODE REVIEW

**Comment by AndyAyersMS** [comment]
@amanasifkhalid PTAL
cc @dotnet/jit-contrib 

No diffs expected. I though there were more possible callers, so perhaps I've missed some. Will also be adding more with the deabstraction work.

**Comment by AndyAyersMS** [comment]
@amanasifkhalid fyi

File: src/coreclr/jit/fgprofile.cpp [hunk 1]
```diff
@@ -5063,3 +5063,115 @@ bool Compiler::fgDebugCheckOutgoingProfileData(BasicBlock* block, ProfileChecks
...
+// Arguments:
+//   block        - block that was just altered
+//   retainedEdge - flow edge that remains
+//   removedEdge  - flow edge that was removed
+//   metric       - [in/out, optional] metric to update if profile becomes inconsistent
+//
+void Compiler::fgRepairProfileCondToUncond(BasicBlock* block,
+                                           FlowEdge*   retainedEdge,
+                                           FlowEdge*   removedEdge,
+                                           int*        metric)
```
Comment (amanasifkhalid) [inline src/coreclr/jit/fgprofile.cpp:5080]:
```suggestion
                                           int*        metric /* = nullptr */)
```

File: src/coreclr/jit/fgprofile.cpp [hunk 2]
```diff
@@ -5063,3 +5063,115 @@ bool Compiler::fgDebugCheckOutgoingProfileData(BasicBlock* block, ProfileChecks
...
+    if (!block->hasProfileWeight())
+    {
+        return;
+    }
+
+    // If the removed edge was not carrying away any profile, there's nothing to do.
+    //
+    weight_t const weight = removedEdge->getLikelyWeight();
+
+    if (weight == 0)
```
Comment (amanasifkhalid) [inline src/coreclr/jit/fgprofile.cpp:5096]: nit:
```suggestion
    if (weight == 0.0)
```
or `BB_ZERO_WEIGHT`

File: src/coreclr/jit/fgprofile.cpp [hunk 3]
```diff
@@ -5063,3 +5063,115 @@ bool Compiler::fgDebugCheckOutgoingProfileData(BasicBlock* block, ProfileChecks
...
+    //
+    BasicBlock* const alternate = removedEdge->getDestinationBlock();
+
+    if (alternate->hasProfileWeight())
+    {
+        weight_t const alternateNewWeight = alternate->bbWeight - weight;
+
+        // If profile weights are consistent, expect at worst a slight underflow.
+        //
+        if (fgPgoConsistent && (alternateNewWeight < 0))
```
Comment (amanasifkhalid) [inline src/coreclr/jit/fgprofile.cpp:5137]:
```suggestion
        if (fgPgoConsistent && (alternateNewWeight < 0.0))
```

File: src/coreclr/jit/fgprofile.cpp [hunk 4]
```diff
@@ -5063,3 +5063,115 @@ bool Compiler::fgDebugCheckOutgoingProfileData(BasicBlock* block, ProfileChecks
...
+        {
+            assert(fgProfileWeightsEqual(alternateNewWeight, 0));
```
Comment (amanasifkhalid) [inline src/coreclr/jit/fgprofile.cpp:5139]:
```suggestion
            assert(fgProfileWeightsEqual(alternateNewWeight, 0.0));
```

**Review by amanasifkhalid — APPROVED** [review]
Sorry I'm late to this. Thanks for adding this in!

**Pull request merged**